### PR TITLE
PLT-6355: Use a separate Read Replica for Search.

### PR DESCRIPTION
**Pull request by grundleborg**
#### Summary
Use a separate Read replica for searching.

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-6355

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has enterprise changes: https://github.com/mattermost/enterprise/pull/136
- [x] Touches critical sections of the codebase (database connection).

### CODE REVIEW

File: store/sql_store.go [hunk 1]
```diff
@@ -110,6 +111,14 @@ func initConnection() *SqlStore {
 		}
 	}
 
+	if len(utils.Cfg.SqlSettings.DataSourceSearchReplica) == 0 {
+		sqlStore.searchReplica = sqlStore.master
```
Comment (coreyhulen) [inline store/sql_store.go]: This should map the the replicas not master when not set.  This also makes me realize maybe this should be an array of search replicas like read replicas?

Comment (grundleborg) [inline store/sql_store.go]: ooh yeah, good point! I'll change it to an array, as I don't think there's any other way of doing it that isn't excessively complicated.

Comment (crspeller) [inline store/sql_store.go]: Make sure when it's blank it doesn't go to a single RR as well. It goes though the regular system.

Comment (grundleborg) [inline store/sql_store.go]: @crspeller should I use a separate round robin always (I think this probably makes more sense), or use the read-replica one when there are no search replicas set?

Comment (crspeller) [inline store/sql_store.go]: I think those are pretty much equivalent. So whichever one works better in the code.

**Comment by grundleborg** [comment]
@coreyhulen updated to address your comments.

**Comment by coreyhulen** [comment]
Looks good.  Once merged we should test this on pre-release.

**Comment by crspeller** [comment]
@grundleborg Needs a rebase

**Pull request merged**